### ISSUE TITLE
[Bug]: Postgres query errors not displaying details in query.data attribute

### ISSUE THREAD

**Issue by felix-appsmith** [appsmithorg/appsmith#21022]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When users run Postgres queries that fail in Appsmith, the response with detailed error information is no longer displayed in the query.data property. Instead, they are only informed that there was a Postgres error and that they need to check the error logs. 

Although error information is still displayed in the error panel, this is preventing users from displaying detailed error messages in the application through a toast message, as they were previously doing. 

This issue is preventing users from understanding and troubleshooting errors efficiently, and is impacting the overall functionality of the application.

It looked like this before (1.8.6): 
![AppsmithBefore](https://user-images.githubusercontent.com/114161539/221693477-904edc42-c424-457f-85f5-f5e83ed4b7d6.png)

It looks like this now (1.9.9):
![appsmithNow](https://user-images.githubusercontent.com/114161539/221693570-6c27b7e7-3a72-4ea2-a312-a696abb727a3.png)

The [documentation](https://docs.appsmith.com/core-concepts/writing-code/workflows#handle-query-success-or-error) specifies the format for the message. If this is a new implementation that will be kept like this, please update the documentation.

### Steps To Reproduce

1. Create a new PostgreSQL datasource.
2. Create a users table with two columns (name and age). 
```bash
CREATE TABLE users (
   name text,
   age text
);
```
3. Create a PostgreSQL query with the name of the age column misspelled as "ageBad".
```bash
INSERT INTO users (name, ageBad) VALUES ('Juan', '30');
```
4. Add a button.
5. Add an OnClick action, checking the JS option.
```bash
{{QueryBad.run(()=> showAlert('"query executed syccessfuly"','success'), () => showAlert(QueryBad.data))
}}
```

### Environment

Production

### Version

Appsmith v1.9.9-SNAPSHOT (CLOUD) 

<img src="https://front.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_7pzbg)

**Comment by mmcallster** [comment]
Hi, I'm having a similar issue with an API datasource.  I used to be able to access the useful error response message in the query.data object.  Now query.data contains a generic message 'An error occurred during the execution of your API. Please check the error logs for more details.'

I can see in the log both errors, an Appsmith error code 5000 and my API error code 400.  I need to display the API reponse error for users to understand what the problem is:
![image](https://user-images.githubusercontent.com/121312970/221976136-e65c92e3-c7aa-4cfd-8443-1db40e132f0e.png)


Thanks

**Comment by hoiast** [comment]
I have the exactly same issue as @mmcallster. I am running appsmith-ee, version 1.9.9 on ECS - Fargate.

**Comment by sribalajig** [comment]
@hoiast @mmcallster this issue has been fixed in `v1.9.10` which will be available shortly. Could you please check the upgraded version and let us know if everything works as before. Thanks.

**Comment by mmcallster** [comment]
Hi @sribalajig,
I'm on v1.9.10 and can confirm the API error message is being set to the query.data object again.  It seems a little bit different, like it's being set as a string instead of an object.  I only say this because my original error handling code (which worked in v1.9.8) needed to modified a little bit to work again in v1.9.10.

All is well though, thanks for the quick patch!
